src/figuresClasses.ts

```diff
@@ -1,11 +1,93 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public aSide: number,
+    public bSide: number,
+    public cSide: number,
+  ) {
+    if (aSide <= 0 || bSide <= 0 || cSide <= 0) {
+      throw new Error('One of entered sides is less or equal 0');
```

Comment:
[CODE STYLE] - When throwing an error, try to provide a detailed description in the error message. For instance, 'One of the triangle sides is less than or equal to 0' would be more descriptive.